backend/app/routers/session.py

```diff
@@ -46,8 +48,21 @@ async def delete_session(
     await session_service.delete_session(db_session, session_id, current_user.id)
     return session_id
 
+
 @router.get("/{session_id}/ranking", response_model=list[UserSessionRankingOut])
 async def get_session_ranking(
     db_session: Annotated[AsyncSession, Depends(get_session)], session_id: str
 ) -> list[UserSessionRankingOut]:
     return await session_service.get_session_ranking(db_session, session_id)
+
+
+@router.post("/{session_id}/ranking/{user_name}/submit", response_model=QuizCheckOut)
+async def check_solution(
+    db_session: Annotated[AsyncSession, Depends(get_session)],
+    session_id: str,
+    user_name: str,
+    quiz_id: str,
```

Comment:
Having session_id we can easly retrieve the quiz_id, so we don't need it here.

Also it'll be treated as a query string parameter, which for me isn't the best way to pass parameters in this case (I'd use request's body)